### PR TITLE
General linting upgrades

### PR DESCRIPTION
**Pull request by devoncarew**
Some general linting upgrades, including adding types to the API boundaries (method params and return types) where possible:

- address a few of the issues reported by `avoid_dynamic_calls`
- address a few of the issues reported by `type_annotate_public_apis`
- run dart fix for prefer_single_quotes

### CODE REVIEW

File: lib/number_symbols_data.dart [hunk 1]
```diff
@@ -22,7 +22,7 @@ library number_symbol_data;
 import "number_symbols.dart";
 
 /// Map from locale to [NumberSymbols] used for that locale.
-Map<dynamic, dynamic> numberFormatSymbols = <String, NumberSymbols>{
+Map<String, NumberSymbols> numberFormatSymbols = {
```
Comment (kevmoo) [inline lib/number_symbols_data.dart:25]: Could this be final?

File: lib/date_symbols.dart [hunk 2]
```diff
@@ -350,7 +350,7 @@ var en_USSymbols = DateSymbols(
       '{1}, {0}'
     ]);
 
-var en_USPatterns = const {
+Map<String, String> en_USPatterns = const {
```
Comment (kevmoo) [inline lib/date_symbols.dart:353]: Could this be final?

File: lib/date_symbols.dart [hunk 3]
```diff
@@ -212,7 +212,7 @@ class DateSymbols {
 
 /// We hard-code the locale data for en_US here so that there's at least one
 /// locale always available.
-var en_USSymbols = DateSymbols(
+DateSymbols en_USSymbols = DateSymbols(
```
Comment (kevmoo) [inline lib/date_symbols.dart:215]: Could this be final?

File: test/plural_test.dart [hunk 4]
```diff
@@ -21,7 +21,7 @@ import 'package:test/test.dart';
 /// use One for the singular, Few for the genitive singular, and Many for the
 /// genitive plural. Other would be used for fractional values if we supported
 /// those.
-var expectedRu = '''
+String expectedRu = '''
```
Comment (kevmoo) [inline test/plural_test.dart:24]: Could this be final? and below?

**Review by kevmoo — APPROVED** [review]

File: lib/number_symbols_data.dart (hunk 1 above)
Comment (devoncarew) [reply lib/number_symbols_data.dart:25]: Yup; done.

File: lib/date_symbols.dart (hunk 2 above)
Comment (devoncarew) [reply lib/date_symbols.dart:353]: Made this `const`.

File: lib/date_symbols.dart (hunk 3 above)
Comment (devoncarew) [reply lib/date_symbols.dart:215]: done

File: test/plural_test.dart (hunk 4 above)
Comment (devoncarew) [reply test/plural_test.dart]: yup; updated.

**Comment by devoncarew** [comment]
Note - this PR has been updated to remove ~2 type annotations; these were making the google3 roll difficult. The annotation rollbacks are tracking in comments in the source as well as in https://github.com/dart-lang/intl/issues/482.

**Pull request merged**